File: forge/src/main/java/juuxel/adorn/platform/forge/compat/ArchitectsPaletteCompat.java
package juuxel.adorn.platform.forge.compat;

import juuxel.adorn.block.variant.BlockVariant;
import juuxel.adorn.block.variant.PrefixedBlockVariantSet;

import java.util.List;

public final class ArchitectsPaletteCompat extends PrefixedBlockVariantSet {
    @Override
    public String getModId() {
        return "architects_palette";
    }

    @Override
    public List<BlockVariant> getWoodVariants() {
        return List.of(new BlockVariant.Wood("architects_palette/twisted"));
    }

    @Override
    public List<BlockVariant> getStoneVariants() {
        return createVariants(
            BlockVariant.Stone::new,
            "myonite",
            "myonite_brick",
            "mushy_myonite_brick"
        );
    }
}

File: forge/src/main/kotlin/juuxel/adorn/platform/forge/compat/ArchitectsPaletteCompat.kt
package juuxel.adorn.platform.forge.compat

import juuxel.adorn.api.block.BlockVariant
import juuxel.adorn.compat.BlockVariantSet

object ArchitectsPaletteCompat : BlockVariantSet {
    override val woodVariants = listOf(BlockVariant.Wood("architects_palette/twisted"))

    override val stoneVariants = listOf(
        "myonite",
        "myonite_brick",
        "mushy_myonite_brick",
    ).map { BlockVariant.Stone("architects_palette/$it") }
}
